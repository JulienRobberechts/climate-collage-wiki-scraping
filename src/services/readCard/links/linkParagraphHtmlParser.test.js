const { getContent } = require('../../utils/fileServices/readFile');
const { parseLinks } = require('./linkParagraphHtmlParser');

const linksEffectsFilePathForCard =
  cardNum => `./src/services/readCard/links/test-data/fr/linksEffects${cardNum}.html`;

describe('parse other links', () => {
  it('parse optional Effects card 1', async () => {
    const content = await getContent(linksEffectsFilePathForCard(1));
    const links = parseLinks(content);
    expect(links).toMatchSnapshot();
    expect(links.length).toBe(2);
    expect(links[0].explanation).toContain("L'Ãªtre humain occupe presque");
    expect(links[0].href).toEqual("/wiki/index.php?title=Fr-fr_adulte_carte_25_biodiversit%C3%A9_terrestre");
  });
  it('parse optional Effects card 20', async () => {
    const content = await getContent(linksEffectsFilePathForCard(20));
    const links = parseLinks(content);
    expect(links.length).toBe(2);
    expect(links).toMatchSnapshot();
  });
  it('parse optional Effects card 35', async () => {
    const content = await getContent(linksEffectsFilePathForCard(35));
    const links = parseLinks(content);
    expect(links.length).toBe(5);
    expect(links).toMatchSnapshot();
  });
});
