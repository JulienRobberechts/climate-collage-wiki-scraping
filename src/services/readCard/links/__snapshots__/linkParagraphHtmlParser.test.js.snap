// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse other links parse optional Causes card 25 1`] = `
Array [
  Object {
    "explanation": "On pense ici surtout aux pesticides (notamment au Round-up ou aux n\\\\u00e9onicotino\\\\u00efdes). Pas de lien avec le climat, mais relation int\\\\u00e9ressante \\\\u00e0 faire",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_8_agriculture\\\\\\"",
  },
  Object {
    "explanation": "La d\\\\u00e9forestation cause \\\\u00e9norm\\\\u00e9ment de perte de biodiversit\\\\u00e9 :\\\\n Animale car la for\\\\u00eat est un h\\\\u00f4te de beaucoup d'esp\\\\u00e8ce. L'indice de biodiversit\\\\u00e9 des for\\\\u00eats a chut\\\\u00e9 de 53% entre 1970 et 2014.\\\\nV\\\\u00e9g\\\\u00e9tale car sur les 60 000 esp\\\\u00e8ces d'arbres connues, 8000 sont consid\\\\u00e9r\\\\u00e9es comme menac\\\\u00e9es.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_6_d%C3%A9forestation\\\\\\"",
  },
  Object {
    "explanation": "Les deux types de biodiversit\\\\u00e9 peuvent interagir entres elles.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_27_biodiversit%C3%A9_marine\\\\\\"",
  },
]
`;

exports[`parse other links parse optional Effects card 1 1`] = `
Array [
  Object {
    "explanation": "L'être humain occupe presque tout l'espace disponible sur terre, ne laissant aucune place aux animaux et aux plantes. C'est ce qu'on appelle la disparition des habitats naturels et c'est la principale cause de perte de biodiversité aujourd'hui, bien devant les causes climatiques.",
    "href": "/wiki/index.php?title=Fr-fr_adulte_carte_25_biodiversit%C3%A9_terrestre",
  },
  Object {
    "explanation": "Par ce lien, on exprime toutes les dégradations que l'être humain est capable d'infliger à la vie marine comme la pollution au plastique et la sur-pêche. C'est hors-sujet par rapport au changement climatique, mais c'est intéressant de faire le lien quand-même.",
    "href": "/wiki/index.php?title=Fr-fr_adulte_carte_27_biodiversit%C3%A9_marine",
  },
]
`;

exports[`parse other links parse optional Effects card 20 1`] = `
Array [
  Object {
    "explanation": "C'est tr\\\\u00e8s technique, mais si on est pointilleux, la partie bleue du graphique de la carte 19 concernant l'Antarctique repr\\\\u00e9sente un gain de masse d\\\\u00fb \\\\u00e0 une augmentation des pr\\\\u00e9cipitations. Les parties rouges repr\\\\u00e9sentent une perte de masse. Au total, l'Antarctique perd de la masse.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_19_fonte_des_calottes_glaciaires\\\\\\"",
  },
  Object {
    "explanation": "La carte vecteurs de maladie est en g\\\\u00e9n\\\\u00e9ral reli\\\\u00e9e \\\\u00e0 la carte Biodiversit\\\\u00e9 terrestre car les vecteurs de maladie sont une sous-partie de la biodiversit\\\\u00e9, mais on peut aussi la relier aux m\\\\u00eames causes que la carte biodiversit\\\\u00e9, c\\\\u2019est-\\\\u00e0-dire perturbation du cycle de l'eau et hausse de la temp\\\\u00e9rature.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_28_vecteurs_de_maladie\\\\\\"",
  },
]
`;

exports[`parse other links parse optional Effects card 35 1`] = `
Array [
  Object {
    "explanation": "Une partie de la déforestation est faite en brûlant la forêt qui peut ensuite dégénérer en feu non-maitrisé. C'est ce qu'il s'est passé pendant l'été 2019 en Amazonie et en Australie",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_6_d%C3%A9forestation\\\\\\"",
  },
  Object {
    "explanation": "Les analyses par satellite montrent dans les fumées des quantités de particules toxiques qui posent des problèmes de santé publique.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_10_a%C3%A9rosols\\\\\\"",
  },
  Object {
    "explanation": "Cela reste un élément mineur. Sauf sur des hotspots particuliers de biodiversité.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_25_biodiversit%C3%A9_terrestre\\\\\\"",
  },
  Object {
    "explanation": "En général, ce qui brûle bien, c'est la forêt, pas les champs de blé.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_32_baisse_rendements_agricoles\\\\\\"",
  },
  Object {
    "explanation": "630 disparus suite aux incendies en Californie à l'automne 2018.",
    "href": "\\\\\\"/wiki/index.php?title=Fr-fr_adulte_carte_38_sant%C3%A9_humaine\\\\\\"",
  },
]
`;
